File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/ui/base/BasePresenter.java
package com.example.ama.android2_lesson01.ui.base;

/**
 * Base implementation of {@link Presenter}
 *
 * @param <T> view with which to work
 */

abstract public class BasePresenter<T extends NoteView> implements Presenter<T> {
    protected T view;

    /**
     * Attach view with which to work
     *
     * @param view for attaching
     */

    @Override
    public void attachView(T view) {
        this.view = view;
    }

    /**
     * Detach using view
     */

    @Override
    public void detachView() {
        view = null;
    }

}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/ui/base/BasePresenter.kt
package com.example.ama.android2_lesson01.ui.base

import com.example.ama.android2_lesson01.model.Note

/**
 * Base implementation of [Presenter]
 *
 * @param <T> view with which to work
 */

abstract class BasePresenter<T : NoteView> : Presenter<T> {
    protected var view: T? = null

    /**
     * Attach view with which to work
     *
     * @param view for attaching
     */

    override fun attachView(view: T) {
        this.view = view
    }

    /**
     * Detach using view
     */

    override fun detachView() {
        view = null
    }
}